### PR TITLE
Unify pre-rust startup code, NMI/Exception entry point  for FMC and RT

### PR DESCRIPTION
**Pull request by rusty1968**
This commit includes the following:

- Consolidates pre-rust code (trap, nmi, startup) and linker descriptor file for FMC and RT in the cpu crate.
- Loads and checks the FHT in both FMC and RT binaries.
- Adds a Makefile for FMC to automate the development flow :  building the FMC and RT binary crates, create the firmware image and check FMC can transition to RT.

### CODE REVIEW

**Review by jhand2 — APPROVED** [review]

**Comment by rusty1968** [comment]
Why github drops approvals if we make any changes?

**Review by FerralCoder — DISMISSED** [review]

**Comment by jhand2** [comment]
> Why github drops approvals if we make any changes?

I think that's a Github setting to ensure that what people approved is what gets merged.

Feel free to tag me when you're ready for another round of review and I'll take another look.

**Comment by rusty1968** [comment]
@jhand2 , ready for another round of review. Vishal recommended we share the minimum amount of code possible with ROM, so I removed the rental cell and the environment type from the common crate.

File: cpu/build.sh
Comment (korran) [inline cpu/build.sh:1]: Is this file used?

**Review by korran — APPROVED** [review]

**Pull request merged**